File: 0917-reverse-only-letters/0917-reverse-only-letters.cpp
class Solution {
public:
    string reverseOnlyLetters(string s) {
        auto left = 0, right = (int)s.size() - 1;
        while (left < right) {
            if (!isalpha(s[left])) left++;
            else if (!isalpha(s[right])) right--;
            else swap(s[left++], s[right--]);
        }
        return s;
    }
};

File: 0917-reverse-only-letters/0917-reverse-only-letters.js
/**
 * @param {string} s
 * @return {string}
 */

var reverseOnlyLetters = function(s) {
    
    let initArr = s.split('')
    let left = 0
    let right = initArr.length - 1
    
    function isLetter(char) {
        return /^[a-zA-Z]$/.test(char);
    }
   
    while (left < right) {
        
        while(left < right && isLetter(initArr[left]) === false) { left ++ }
        while(left < right && isLetter(initArr[right]) === false) { right-- }
        if (left >= right) break;
        
        let temp = initArr[left]
        initArr[left] = initArr[right]
        initArr[right] = temp
        left++
        right--
    }
    
    return initArr.join('')
};